hack/clean/clean.go

```diff
@@ -31,6 +31,7 @@ var denylist = []string{
 	"images",
 	"secrets",
 	"dns",
+	"shared-cluster",
```

Comment:
The shared cluster's managed RG should be safe since it doesn't have a `purge` tag, but you could add it here just for extra safety if you think it's worth it (`aro-dsfx17my`).

Reply:
I could do that, but I believe that the deny assignment + the lack of purge tag should protect us here. Plus, if we ever had to reinstall the shared cluster, the managed resource group ID would change, and we'd need a new PR.